File: backend/lib/event.js
const axiosLib = require("axios");
const fs = require("fs");

const WILCO_ID = process.env.WILCO_ID || fs.readFileSync('../.wilco', 'utf8')

const baseURL = 'https://engine.wilco.gg';
const axios = axiosLib.create({
  baseURL: baseURL,
  headers: {
    'Content-type': 'application/json',
  },
});

async function sendEvent(event, metadata) {
  const result = await axios.post(`/users/${WILCO_ID}/event`, JSON.stringify({ event, metadata }));
  return result.data;
}

module.exports = {
  sendEvent,
}

File: backend/app/services/event.py
import os
import requests
import json

PATH_TO_WILCO_ID = '../.wilco'
BASE_URL = 'https://engine.wilco.gg'
WILCO_ID = os.environ.get('WILCO_ID')

if not WILCO_ID and os.path.exists(PATH_TO_WILCO_ID):
    with open(PATH_TO_WILCO_ID, 'r') as f:
        WILCO_ID = f.read()

EVENTS_ENDPOINT = f'{BASE_URL}/users/{WILCO_ID}/event'

def send_event(event, metadata):
    headers = { 'Content-type': 'application/json' }
    data = { 'event': event, 'metadata': metadata }

    res = requests.post(EVENTS_ENDPOINT, data=json.dumps(data), headers=headers)
    return res
